frontend/src/components/MeetingRoom/PinButton/PinButton.tsx

```diff
@@ -0,0 +1,98 @@
+import PushPinIcon from '@mui/icons-material/PushPin';
+import { IconButton, Tooltip } from '@mui/material';
+import { MouseEvent, ReactElement, useRef, useState } from 'react';
+import PushPinOffIcon from '../../Icons/PushPinOffIcon';
+import isMouseEventInsideBox from '../../../utils/isMouseEventInsideBox';
+
+export type PinButtonProps = {
+  color?: string;
+  isMaxPinned: boolean;
+  isPinned: boolean;
+  isTileHovered: boolean;
+  participantName?: string;
+  pinStyle?: string;
+  handleClick: (clickEvent: MouseEvent<HTMLButtonElement>) => void;
+};
+
+const PinButton = ({
+  color,
+  isMaxPinned,
+  isPinned,
+  isTileHovered,
+  participantName,
+  pinStyle,
+  handleClick,
+}: PinButtonProps): ReactElement | false => {
+  const isDisabled = isMaxPinned && !isPinned;
+  const anchorRef = useRef<HTMLDivElement | null>(null);
+  const [isHoveringButton, setIsHoveringButton] = useState<boolean>(false);
+  const iconSx = {
+    fontSize: '18px',
+    color: isDisabled ? 'rgba(255,255,255,.54)' : color,
+    cursor: 'pointer',
+  };
+
+  const getTooltipText = () => {
+    if (isDisabled) {
+      return `You can't pin any more tiles`;
```

Comment:
nit: maybe instead of a current message, we could say the `Current ${MAX_PIN} pin limit has been reached` or something like that?

Reply:
This text was more inline with our UX Guide